File: ischia/src/test/java/cz/fb/manaus/ischia/BackLoserBettorTest.java
package cz.fb.manaus.ischia;

import cz.fb.manaus.core.model.Side;
import cz.fb.manaus.reactor.betting.AbstractBettorTest;
import org.junit.Test;
import org.springframework.test.context.ActiveProfiles;

import java.util.OptionalDouble;

import static java.util.OptionalDouble.of;

@ActiveProfiles("ischia")
public class BackLoserBettorTest extends AbstractBettorTest<BackLoserBettor> {

    @Test
    public void testPlaceBest() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.98d, 3.8d, of(3d), 1)),
                3, OptionalDouble.of(3.75d));
    }

    @Test
    public void testPlaceFairness() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.98d, 3.3d, of(3d), 1)),
                3, OptionalDouble.of(3.25d));
    }

    @Test
    public void testLastMatchedOrTradedVolume() throws Exception {
        checkPlace(persistMarket(reactorTestFactory.createMarket(2.98d, 3.2d, of(4d), 1)),
                3, OptionalDouble.of(4.0));
    }

    @Test
    public void testCloseUpdate() {
        var market = persistMarket(reactorTestFactory.createMarket(2.8d, 3.4d, of(3d), 1));
        checkUpdate(market, 3.4d, Side.BACK, 0, 0);
        checkUpdate(market, 3.35d, Side.BACK, 0, 0);
        checkUpdate(market, 3.5d, Side.BACK, 0, 3);
        checkUpdate(market, 3.3d, Side.BACK, 0, 3);
        checkUpdate(market, 3.25d, Side.BACK, 0, 3);
    }

}

File: ischia/src/test/java/cz/fb/manaus/ischia/BackLoserBettorTest.kt
package cz.fb.manaus.ischia

import cz.fb.manaus.core.model.Side
import cz.fb.manaus.core.test.AbstractIntegrationTestCase
import cz.fb.manaus.reactor.PricesTestFactory
import cz.fb.manaus.reactor.betting.BettorTester
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.ActiveProfiles

@ActiveProfiles("ischia")
class BackLoserBettorTest : AbstractIntegrationTestCase() {

    @Autowired
    private lateinit var bettorTester: BettorTester

    @Test
    fun `place bet - based on best price`() {
        val marketPrices = PricesTestFactory.newMarketPrices(2.98, 3.8, 3.0)
        bettorTester.checkPlace(Side.BACK, marketPrices, 3, 3.772)
    }

    @Test
    fun `place bet - based on fairness`() {
        val marketPrices = PricesTestFactory.newMarketPrices(2.98, 3.1, 3.0)
        bettorTester.checkPlace(Side.BACK, marketPrices, 3, 3.151)
    }

    @Test
    fun `update bet`() {
        val market = PricesTestFactory.newMarketPrices(2.8, 3.4, 3.0)
        bettorTester.checkUpdate(Side.BACK, 3.0, market, 0, 3)
    }

}
